### PR TITLE
Refactor tests and allow searching with new type

### PR DESCRIPTION
**Pull request by JamesJeffryes**
The code with now also index features in the mrnas, cdss and non_codeing_features arrays as well as deal with braking changes to the alias, function and ontology fields. This PR updates the tests to use local genomes of old an new types.

### CODE REVIEW

**Comment by JamesJeffryes** [comment]
@briehl Thanks for the review

File: GenomeSearchUtil.spec [hunk 1]
```diff
@@ -40,8 +40,10 @@ module GenomeSearchUtil {
             (value),
         global_location - this is location-related properties that are
             under sorting whereas items in "location" array are not,
-        feature_idx - legacy field keeping the position of feature in
-            feature array in legacy Genome object,
+        feature_array - field recording which array a feature is located in
+            (features, mrnas, cdss, ect.)
+        feature_idx - field keeping the position of feature in
+            it's array in a Genome object,
```
Comment (briehl) [inline GenomeSearchUtil.spec]: it's -> its (yeah, I know I'm being pedantic...)

File: kbase.yml [hunk 2]
```diff
@@ -8,10 +8,10 @@ service-language:
     python
 
 module-version:
-    0.0.3
+    0.0.4
 
 owners:
-    [rsutormin, msneddon]
+    [scannon, sychan, jjeffryes]
```
Comment (briehl) [inline kbase.yml]: scannon -> scanon

File: GenomeSearchUtil.spec (hunk 1 above)
Comment (briehl) [reply GenomeSearchUtil.spec]: Also, is there a list of allowed names? If not, all of those (not just a sample) should be here.

File: lib/GenomeSearchUtil/GenomeSearchUtilIndexer.py [hunk 3]
```diff
@@ -135,11 +154,15 @@ def check_feature_cache(self, ref, token):
...
                 print("    Loading WS object...")
             t1 = time.time()
-            genome = ws_client.get_objects2({"objects": [{"ref": ref, "included": [
-                    "/features/[*]/id", "/features/[*]/type", "/features/[*]/function", 
-                    "/features/[*]/aliases", "/features/[*]/location",
-                    "/features/[*]/ontology_terms/*/*/term_name"]}]})["data"][0]["data"]
-            self.save_feature_tsv(genome["features"], inner_chsum)
+            incl = [x+y for x,y in product(
+                ['features/[*]/', 'cdss/[*]/', 'mrnas/[*]/',
+                 'non_codeing_features/[*]/'],
```
Comment (briehl) [inline lib/GenomeSearchUtil/GenomeSearchUtilIndexer.py]: non_codeing_features -> non_coding_features (I think, that's what's used above, at least)

**Review by briehl — CHANGES_REQUESTED** [review]
Looks pretty good, some minor tweaks to names.
However, tests are failing, since the GenomeSearchUtil_server_test module can't find the AssemblyUtil. Maybe it didn't get added in your commit?

**Comment by JamesJeffryes** [comment]
@briehl @sychan I've addressed Bill's comments and added the Assembly Utils dependancy. If the tests run can you merge & register the app?

**Review by briehl — APPROVED** [review]

**Pull request merged**